Data Structures/BinaryTree/BT_CountnonLeafNodes.java

```diff
@@ -0,0 +1,118 @@
+
+import java.util.Scanner;
+
+public class BT_CountnonLeafNodes {
+
+	static Scanner scn = new Scanner(System.in);
+
+	public static void main(String[] args) {
+
+		BT_CountnonLeafNodes bts = new BT_CountnonLeafNodes();
+		BinaryTree bt = bts.new BinaryTree();
+		// 10 true 20 true 40 false false true 50 false false true 30 true 60 false
+		// false false **Give this Input to create the tree**
+
+		System.out.println(bt.countNonLeafNodes());
+
+	}
+
+	private class BinaryTree {
+
+		private class Node {
+			int data;
+			Node left;
+			Node right;
+		}
+
+		private Node root;
+		private int size;
+
+		public BinaryTree() {
+			this.root = construct(null, false);
+		}
+
+		private Node construct(Node parent, boolean ilc) {	// ilc -> is left child (give true for left and false for right)
+
+			if (parent == null) {
+				System.out.println("Enter the data for root node ? ");
+			} else {
+				if (ilc) {
+					System.out.println("Enter the data for left " + parent.data);
+				} else {
+					System.out.println("Enter the data for right " + parent.data);
+				}
+			}
+
+			int val = scn.nextInt();
+			Node nn = new Node();	// nn -> new node 
+			nn.data = val;
+			this.size++;
+
+			boolean lc;		// lc -> left child
+			System.out.println("Do you want left child");

```

Comment:
Split the code into addLeftChild and addRightChild functions? Even better is addChild(position).
You are duplicating a lot of code.